tests/activity_test.go

```diff
@@ -346,6 +348,7 @@ func (s *functionalSuite) TestActivityRetry() {
 
 	descResp, err := describeWorkflowExecution()
 	s.NoError(err)
+	// assert that all the activities with Id == "A" run failed and were run by poller[1]
```

Comment:
both `poller` and `poller2` have `Identity == identity == worker1`, so this actually asserts that the activities were run by `worker1`, not either of the pollers

Reply:
Ha! You are right. I did not notice this when I was reading the test. There were no need for 2 pollers in the test in the first place. Having them have the same identity actually make them the same poller from the server point of view.